backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/MigrateVmCommand.java

```diff
@@ -418,6 +418,19 @@ private MigrateVDSCommandParameters createMigrateVDSCommandParameters() {
                 getCluster().getMigrationPolicyId(),
                 getCluster().getCompatibilityVersion());
         MigrationPolicy effectiveMigrationPolicy = findEffectiveConvergenceConfig(clusterMigrationPolicy);
+
+        boolean zerocopy = false;
+        if (effectiveMigrationPolicy.isZerocopy()
+                && FeatureSupported.isParallelMigrationsSupported(getVm().getCompatibilityVersion())) {
+            migrateCompressed = false;
+            if (parallelMigrations == null || parallelMigrations == -2 || parallelMigrations == 0) {
+                parallelMigrations = -1;
+            }
+            if (!migrateEncrypted) {
```

Comment:
And if `migrateEncrypted` is true, so we don't enable zerocopy migration, are the other changes that we just made (`migrateCompressed  = false`, `parallelMigrations = -1`) still relevant?

Reply:
I don't think so, good catch.  I'll fix it.  I also planned to replace the -2, -1, 0 parallel migration constants with symbols, I'll do it too.

Reply:
Done.  (I think the original idea was based on "If encrypted migrations are requested, the given policy
can still be used but zero-copy is disabled." but it doesn't make sense in this case.)